ansible/roles/satellite-manage-repositories/tasks/version_6.x.yml

```diff
@@ -0,0 +1,51 @@
+---
+- name: Setting up satellite repository
+  theforeman.foreman.katello_repository_set:
+    username: "{{ satellite_admin }}"
+    password: "{{ satellite_admin_password }}"
+    server_url: "https://{{ publicname }}"
+    validate_certs: no
+    organization: "{{ org }}"
+    name: "{{ item.1.name }}"
+    product: "{{ item.1.product }}"
+    repositories:
+    - releasever: "{{ item.1.releasever | default(omit) }}"
+      basearch: "{{ item.1.basearch | default(omit) }}"
+  loop: "{{ satellite_content | subelements('repos') }}"
+  tags:
+    - configure_satellite
+    - configure_satellite_reposet
+
+## Sync repositories
+- name: Sync repository
+  theforeman.foreman.katello_sync:
+    username: "{{ satellite_admin }}"
+    password: "{{ satellite_admin_password }}"
+    server_url: "https://{{ publicname }}"
+    validate_certs: no
+    organization: "{{ org }}"
+    product: "{{ item.1.product }}"
+    repository: "{{ item.1.name | regex_replace('[()]')  + (' ' + item.1.basearch if item.1.basearch|d(None) else '' ) + ( ' ' + item.1.releasever if item.1.releasever|d(None)  else '' ) }}"
+  loop: "{{ satellite_content | subelements('repos') }}"
+  async: 999999
+  poll: 0
+  register: repo_sync_sleeper
+  when: item.1.sync | default(True) | bool
+  tags:
+    - configure_satellite
+    - configure_satellite_sync
+
+- name: Wait until all Syncs have finished
+  async_status:
+    jid: "{{ repo_sync_sleeper_item.ansible_job_id }}"
+  loop: "{{ repo_sync_sleeper.results }}"
+  loop_control:
+    loop_var: repo_sync_sleeper_item
+  when: repo_sync_sleeper_item.ansible_job_id is defined  # Skip items that were skipped in the previous task
+  register: async_job_result
+  until: async_job_result.finished
```

Comment:
Why do you detach ?

Usually in ansible we detach if we plan to do other tasks in between and join later to wait for the background tasks.

Here you can just run with async (because it's a very long task, so it's required) but without the extra wait by not specifying `poll: 0`, or specify `poll: 60` or whatever reasonable value.


Please use a readable format for async timeout:

```
          async: "{{ 2 * 60 * 60 }}"

```

Reply:
Here I detach, because I want to sync in parallel and then have a semaphore to wait.
If I cycle the async, I'll sync the repositories in serial, right?

Reply:
ah i missed that, good!